deploy/template.yaml

```diff
@@ -187,7 +187,20 @@ Resources:
         ServerSideEncryptionConfiguration:
           - ServerSideEncryptionByDefault:
               SSEAlgorithm: AES256
-      BucketName: !Sub ${AWS::StackName}-test-results
+      BucketName:
+        Fn::Join:
+            - '-'
+            - - Ref: AWS::StackName
```

Comment:
does Ref work without the !

Reply:
Yes you can do `Ref:` or `!Ref`

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html#YAML